provider-example/src/aead.rs

```diff
@@ -22,6 +22,10 @@ impl cipher::Tls13AeadAlgorithm for Chacha20Poly1305 {
     fn key_len(&self) -> usize {
         chacha20poly1305::ChaCha20Poly1305::key_size()
     }
+
+    fn extract_keys(&self, _key: AeadKey, _iv: Iv) -> ConnectionTrafficSecrets {
```

Comment:
How about having the fallibility commit first and then making it required, so that we can have these emit an error directly?

Reply:
Re-ordered to put the fallibility and `ConnectionTrafficSecrets` changes ahead of everything else.

> so that we can have these emit an error

Once re-ordered since `Chacha20Poly1305` is supported by `ConnectionTrafficSecrets` we don't need to emit an error, just implement the functionality using `ConnectionTrafficSecrets::slice_to_array`.